### PR TITLE
Create invoice v1.0

### PR DESCRIPTION
**Pull request by SBASHEET**

### CODE REVIEW

**Comment by siddhi** [comment]
Some tests are failing
```
com.checkout.acceptance.CheckoutAcceptanceTest > shouldThrowExceptionIfInValidBookId FAILED
    java.lang.Exception
        Caused by: org.springframework.web.client.ResourceAccessException at CheckoutAcceptanceTest.java:68
            Caused by: java.net.ConnectException at CheckoutAcceptanceTest.java:68

com.checkout.acceptance.CheckoutAcceptanceTest > shouldCreateInvoiceForValidBookId FAILED
    org.springframework.web.client.ResourceAccessException at CheckoutAcceptanceTest.java:46
        Caused by: java.net.ConnectException at CheckoutAcceptanceTest.java:46
```

**Comment by SBASHEET** [comment]
> Some tests are failing
> 
> ```
> com.checkout.acceptance.CheckoutAcceptanceTest > shouldThrowExceptionIfInValidBookId FAILED
>     java.lang.Exception
>         Caused by: org.springframework.web.client.ResourceAccessException at CheckoutAcceptanceTest.java:68
>             Caused by: java.net.ConnectException at CheckoutAcceptanceTest.java:68
> 
> com.checkout.acceptance.CheckoutAcceptanceTest > shouldCreateInvoiceForValidBookId FAILED
>     org.springframework.web.client.ResourceAccessException at CheckoutAcceptanceTest.java:46
>         Caused by: java.net.ConnectException at CheckoutAcceptanceTest.java:46
> ```



> Some tests are failing
> 
> ```
> com.checkout.acceptance.CheckoutAcceptanceTest > shouldThrowExceptionIfInValidBookId FAILED
>     java.lang.Exception
>         Caused by: org.springframework.web.client.ResourceAccessException at CheckoutAcceptanceTest.java:68
>             Caused by: java.net.ConnectException at CheckoutAcceptanceTest.java:68
> 
> com.checkout.acceptance.CheckoutAcceptanceTest > shouldCreateInvoiceForValidBookId FAILED
>     org.springframework.web.client.ResourceAccessException at CheckoutAcceptanceTest.java:46
>         Caused by: java.net.ConnectException at CheckoutAcceptanceTest.java:46
> ```

Because i have used localhost?

**Comment by siddhi** [comment]
It gives this error when starting the service

```
***************************
APPLICATION FAILED TO START
***************************

Description:

Parameter 1 of constructor in com.checkout.services.CheckoutService required a bean of type 'org.springframework.web.client.RestTemplate' that could not be found.


Action:

Consider defining a bean of type 'org.springframework.web.client.RestTemplate' in your configuration.
```

File: src/test/java/com/checkout/acceptance/CheckoutAcceptanceTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package com.checkout.acceptance;
+
+//@RunWith(SpringRunner.class)
+//@SpringBootTest
+//@AutoConfigureMockMvc
+public class CheckoutAcceptanceTest {
```
Comment (siddhi) [inline src/test/java/com/checkout/acceptance/CheckoutAcceptanceTest.java:6]: Is this test not required anymore?

Comment (cdineshs-ford) [reply src/test/java/com/checkout/acceptance/CheckoutAcceptanceTest.java:6]: this class needs more changes Sid we have changed the approach, as we were running out of the time we committed

Comment (siddhi) [reply src/test/java/com/checkout/acceptance/CheckoutAcceptanceTest.java:6]: Ok, now that we have till next wednesday can the changes be done and committed, then I will

Comment (cdineshs-ford) [reply src/test/java/com/checkout/acceptance/CheckoutAcceptanceTest.java:6]: ok will do changes and commit again

**Pull request merged**